src/type-translator.ts

```diff
@@ -488,18 +488,17 @@ export class TypeTranslator {
         return '?';
       }
 
-      if (type.symbol.flags & ts.SymbolFlags.TypeLiteral) {
-        return this.translateTypeLiteral(type);
-      } else if (
-          type.symbol.flags & ts.SymbolFlags.Function ||
+      if (type.symbol.flags & ts.SymbolFlags.Function ||
           type.symbol.flags & ts.SymbolFlags.Method) {
         const sigs = this.typeChecker.getSignaturesOfType(type, ts.SignatureKind.Call);
         if (sigs.length === 1) {
           return this.signatureToClosure(sigs[0]);
         }
+        this.warn('unhandled anonymous type with multiple call signatures');
+        return '?';
+      } else {
+        return this.translateTypeLiteral(type);
```

Comment:
Is there any other case (other than function, method, type literal)?  Maybe it's worth making this an 'else if' to allow an 'else' bailout?

Reply:
All object types are essentially the empty type plus some properties defined on it. We special case `function` and `method` here because we know how to emit call signatures in a way that fits Closure, and we special case construct signatures inside of translate below for the same reason (I suspect we could simplify this, e.g. shouldn't all functions in the end of the day be types with just one call signature? we handle that below in translate).

So I believe that translateTypeLiteral actually implements an arbitrary translateAnonymousType. We still fall back inside of it if we can't figure out the properties. I think that should be enough?